### ISSUE TITLE
Add Semantic Versioning tags to repository

### ISSUE THREAD

**Issue by dataclouder** [vmware/go-vcloud-director#139]
The current tag for the repository is `v1.0.0`, which is outdated.

As we have introduced new functionalities and some incompatibilities, we need to tag the current repo as `v2.0.0`, as the basis for terraform-provider-vcd.

After the tag is established, it must be updated at every change:

* If there are no API changes (i.e. bug fixing, code cleanup, etc), we increase the revision number (e.g. from `v2.0.5` to `v2.0.6`).

* If we introduce a new feature, without breaking compatibility with the previous version, we increment the minor number (e.g. from `v2.0.6` to `v2.1.0`).

* If we break compatibility, meaning that one or more **public** identifiers or signatures are changed, we need to bump up the major number (e.g. from `v2.1.0` to `v3.0.0`).

These tags are needed to simplify the import of modules from `terraform-provider-vcd`, which should be upgraded to use Go 1.11 and switch from using the `vendor` folder to leveraging [`go modules`](https://github.com/golang/go/wiki/Modules).

**Comment by lvirbalas** [comment]
> If we introduce a new feature, without breaking compatibility with the previous version, we increment the minor number (e.g. from v2.0.6 to v2.1.0).

Bumping this up with each commit seems overwhelming, given the amount of work being put in.

> If we introduce a new feature, without breaking compatibility with the previous version, we increment the minor number (e.g. from v2.0.6 to v2.1.0).

Same comment as above, just as a sample we may introduce ten new features pretty quickly. Would this mean we end up having 2.10.0?

> If we break compatibility, meaning that one or more public identifiers or signatures are changed, we need to bump up the major number (e.g. from v2.1.0 to v3.0.0).

In general, I agree, but I disagree that this should happen instantly after commit. We should "accumulate" some changes and only then increase the major number, otherwise we'll end up with v42.0.0 in no time.

**Comment by dataclouder** [comment]
> We should  "accumulate" some changes and only then increase the major number, otherwise we'll end up with v42.0.0 in no time.

This is not how the system works. A major number increase happens only on incompatible changes, which we should try to avoid. The recommendation of the Go community on the usage of modules is to change the import path for an incompatible change (from `v.1.x` to `v2.0`). Therefore, a `v42.0.0` is very unlikely, while a perfectly acceptable accumulation of changes could result in `v2.42.0`, meaning that this tag can be used by someone who is only using functions introduced in `v2.0.0`.

The change **must** happen at the moment we make the API incompatible, to avoid breaking applications that depend on the previous version tag.

**Comment by lvirbalas** [comment]
> a perfectly acceptable accumulation of changes could result in v2.42.0, meaning that this tag can be used by someone who is only using functions introduced in v2.0.0.

Please rephrase :)

> The change must happen at the moment we make the API incompatible

Usually, we deprecate first before removing/changing signature. Please confirm, we can "accumulate" deprecations, so not to bump the major version? Once ready, we remove deprecated things and bump the number.

**Comment by dataclouder** [comment]
What I mean is this: if a client imports `v2.0.0`, its code should work without changes when our library changes to `v2.42.0`, because we guarantee backwards compatibility as long as the major number is unchanged.

The deprecation is not in contention here. So far, we have introduced incompatibilities when we changed function signatures that breaks depending code. We have done so liberally because we were working towards the implementation of the initial code base. If we pledge to `v2.0.x.x`, we need to stick to the rules and not make incompatible changes. If we do, we must upgrade the version accordingly.
